apps/angular/permissions/src/app/has-permission.guard.ts

```diff
@@ -0,0 +1,44 @@
+import { inject, Injectable } from '@angular/core';
+import { CanMatchFn, Route, Router, UrlTree } from '@angular/router';
+
+import { Role } from './user.model';
+import { UserStore } from './user.store';
+
+@Injectable({ providedIn: 'root' })
+export class HasPermissionService {
+  private readonly router = inject(Router);
+  private readonly userStore = inject(UserStore);
+
+  canMatch(route: Route): boolean | UrlTree {
+    const isAdmin: boolean = route.data?.['isAdmin'] ?? false;
```

Comment:
The goal of using a function is to avoid this hack. use function parameters instead of using the data property of the route.

Reply:
yes you're right, updated